src/main/java/org/broadinstitute/hellbender/tools/exome/alleliccount/AllelicCountWithPhasePosteriors.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.hellbender.tools.exome.alleliccount;
+
+import org.broadinstitute.hellbender.tools.exome.allelefraction.AlleleFractionModeller;
+import org.broadinstitute.hellbender.tools.exome.alleliccount.AllelicCountTableColumn.AllelicCountTableVerbosity;
+import org.broadinstitute.hellbender.utils.MathUtils;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+/**
+ * Represents an {@link AllelicCount} with posterior probabilities for each site being ref minor, alt minor,
+ * or an outlier, according to a model fit by {@link AlleleFractionModeller}.
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ */
+public final class AllelicCountWithPhasePosteriors extends AllelicCount {
+    private final double refMinorProb;
+    private final double altMinorProb;
+    private final double outlierProb;
+
+    /**
+     * Construct the {@link AllelicCountWithPhasePosteriors} object.  If unnormalized probabilities are passed,
+     * they will be normalized.
+     * @param count         {@link AllelicCount} of any verbosity (see {@link AllelicCountTableVerbosity})
+     * @param refMinorLogProb  log posterior probability of ref-minor phase (can be unnormalized)
+     * @param altMinorLogProb  log posterior probability of alt-minor phase (can be unnormalized)
+     * @param outlierLogProb   log posterior probability of outlier phase (can be unnormalized)
+     */
+    public AllelicCountWithPhasePosteriors(final AllelicCount count,
+                                           final double refMinorLogProb, final double altMinorLogProb, final double outlierLogProb) {
+        super(Utils.nonNull(count, "AllelicCount cannot be null."));
+        ParamUtils.isFinite(refMinorLogProb, "Cannot construct AllelicCountWithPhasePosteriors with non-finite ref-minor probability at: " + count.getInterval());
```

Comment:
Note that the error messages in these calls to `isFinite` are not string literals and therefore require new allocation and computation of a `String` even when the check passes.  It's probably not going to harm this tool for an exome but I could imagine bad things for a genome with a lot of hets.  Then again, maybe not.

In gatk public I recently dealt with this by defining eg `Utils.validateArg(final boolean condition, final Supplier<String> message)`, invoked as  `Utils.validateArg(x > 0, () -> x + "is a bad value")`.  I'll probably try to clean up all of gatk protected along these lines some time soon, so I won't demand that you do it now, but it's something to be aware of.